python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-sessions.j2

```diff
@@ -8,17 +8,20 @@
 !
 {%     for monitor_session in monitor_sessions | arista.avd.natural_sort('name') %}
 {%         if monitor_session.access_group.type is arista.avd.defined and monitor_session.access_group.name is arista.avd.defined %}
+{%             set session_ag = true %}
 monitor session {{ monitor_session.name }} {{ monitor_session.access_group.type }} access-group {{ monitor_session.access_group.name }}
 {%         endif %}
 {%         for source in monitor_session.sources | arista.avd.natural_sort('name') %}
 {%             set source_cli = "monitor session " ~ monitor_session.name ~ " source " ~ source.name %}
 {%             if source.direction is arista.avd.defined %}
 {%                 set source_cli = source_cli ~ " " ~ source.direction %}
 {%             endif %}
-{%             if source.access_group.type is arista.avd.defined and source.access_group.name is arista.avd.defined %}
-{%                 set source_cli = source_cli ~ " " ~ source.access_group.type ~ " access-group " ~ source.access_group.name %}
-{%                 if source.access_group.priority is arista.avd.defined %}
-{%                     set source_cli = source_cli ~ " priority " ~ source.access_group.priority %}
+{%             if session_ag | arista.avd.default(false) is arista.avd.defined(false) %}
```

Comment:
This could also be combined with the inner `if` instead of an extra nested level.
```suggestion
{%             if session_ag is not arista.avd.defined %}
```

Reply:
Fixed.